src/cortxfs/cortxfs_fops.c

```diff
@@ -159,17 +159,25 @@ static inline ssize_t __cfs_write(struct cfs_fs *cfs_fs, cfs_cred_t *cred,
 				  size_t count, off_t offset)
 {
 	int rc;
-	struct stat stat;
 	dstore_oid_t oid;
+	struct stat *stat = NULL;
+	struct cfs_fh *fh = NULL;
 	struct dstore *dstore = dstore_get();
-	struct kvnode node = KVNODE_INIT_EMTPY;
 	struct dstore_obj *obj = NULL;
 
+	dassert(cfs_fs && cred && fd && buf);
 	dassert(dstore);
 
+	/* TODO:Temp_FH_op - to be removed
+	 * Should get rid of creating and destroying FH operation in this
+	 * API when caller pass the valid FH instead of inode number
+	 */
+	RC_WRAP_LABEL(rc, out, cfs_fh_from_ino, cfs_fs, &fd->ino, &fh);
+	stat = cfs_fh_stat(fh);
+
 	RC_WRAP_LABEL(rc, out, cfs_ino_to_oid, cfs_fs, &fd->ino, &oid);
 
-	RC_WRAP(cfs_access, cfs_fs, cred, &fd->ino, CFS_ACCESS_WRITE);
+	RC_WRAP_LABEL(rc, out, cfs_access_check, cred, stat, CFS_ACCESS_WRITE);
 
 	if (count == 0) {
```

Comment:
Can we move this to be the first check after the assert.Because if the count is zero, there is no reason to do the access check?

Reply:
Done at 7e81fee7c47c2e7764f37872343de54038817174, as the change set was very minimal have squashed the commits as well.